src/frontend/screens/Library/components/SteamInstall/SteamInstallButton.tsx

```diff
@@ -0,0 +1,128 @@
+import { faSteam } from '@fortawesome/free-brands-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ContextProvider from 'frontend/state/ContextProvider'
+import { useTranslation } from 'react-i18next'
+import './index.scss'
+
+import SteamInstallDialog from './SteamInstallDialog'
+import { launch } from 'frontend/helpers'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { hasStatus } from 'frontend/hooks/hasStatus'
+import { Tooltip } from '@mui/material/'
+
+export default function SteamInstallButton({
+  dataTourId
+}: {
+  dataTourId?: string
+}) {
+  const { showDialogModal, sideloadedLibrary } = useContext(ContextProvider)
+  const { t } = useTranslation()
+  const [showInstallDialog, setShowInstallDialog] = useState(false)
+
+  const { status } = hasStatus('steam')
+  const isLaunching = status === 'playing'
+
+  const isSteamInstalled =
+    sideloadedLibrary.find((app) => app.app_name === 'steam')?.is_installed ||
+    false
+
+  const wineList = useQuery({
+    queryKey: ['alternativeWine'],
+    queryFn: async () => {
+      const response = await window.api.getAlternativeWine()
+      if (!response) return []
+      return response
+    }
+  })
+
+  const isCompatibilityLayerAvailable =
+    wineList.data && wineList.data.length > 0
+
+  const installSteamMutation = useMutation({
+    mutationKey: ['steamInstall'],
+    onSuccess: () => {
+      setShowInstallDialog(false)
+    },
+    mutationFn: async () => window.api.installSteamWindows()
+  })
+
+  const isInstalling = installSteamMutation.isPending
+
+  async function handleSteamInstallation() {
+    if (isSteamInstalled) {
+      return launch({
+        appName: 'steam',
+        t,
+        runner: 'sideload',
+        showDialogModal,
+        hasUpdate: false
+      })
+    }
+
+    return setShowInstallDialog(true)
+  }
+
+  function renderButtonText() {
+    if (isLaunching) {
+      return t('Launching Steam', 'Launching Steam...')
+    }
+    if (isInstalling) {
+      return t('Installing Steam', 'Installing Steam...')
+    }
+    if (isSteamInstalled) {
+      return t('Launch Steam', 'Launch Steam')
+    }
+    return t('Install Steam', 'Install Steam')
+  }
+
+  const toolTipText = isSteamInstalled
+    ? t(
+        'steam-install.launch-tooltip',
+        'Launch Steam using the Heroic Compatibility Layer'
+      )
+    : t(
+        'steam-install.tooltip',
+        `This will install the Windows version of Steam and it will run using a Heroic Compatibility Layer. \n
+    Some games especially older titles, newer 3D games, or those with anti-cheat software may not work properly. \n
+    The Recommended specs are a Mac with Apple Silicon CPU (M1 or above) and at least 16GB of RAM.`
+      )
+
+  const compatibilityLayerNotAvailableTooltip = t(
+    'steam-install.compatibility-layer-not-available',
+    `Compatibility layer not available. Please install one from "Settings > Wine Manager" first.`
```

Comment:
I was wondering, can this actually happen? we auto install one at boot if none is available

I can only imagine a user uninstalling all wines while heroic is running to end up in that state

or heroic failing to install a wine during boot